warehouse/query-core/src/main/java/datawave/query/tables/ScannerFactory.java

```diff
@@ -72,52 +76,75 @@ public ScannerFactory(AccumuloClient client) {
     }
 
     public ScannerFactory(AccumuloClient client, int queueSize) {
+        this.config = null;
+        this.settings = null;
+        this.accrueStats = false;
         try {
             this.cxn = client;
-            scanQueue = new ResourceQueue(queueSize, client);
+            this.scanQueue = new ResourceQueue(queueSize, client);
         } catch (Exception e) {
             throw new RuntimeException(e);
         }
     }
 
-    public synchronized Scanner newSingleScanner(String tableName, Set<Authorizations> auths, Query query) throws TableNotFoundException {
-        if (open) {
+    public Scanner newSingleScanner(String tableName, Set<Authorizations> auths, Query query) throws TableNotFoundException {
+        if (open.get()) {
             Scanner bs = QueryScannerHelper.createScannerWithoutInfo(cxn, tableName, auths, query);
             log.debug("Created scanner " + System.identityHashCode(bs));
             if (log.isTraceEnabled()) {
                 log.trace("Adding instance " + bs.hashCode());
             }
 
-            return bs;
+            synchronized (open) {
+                if (open.get()) {
+                    return bs;
```

Comment:
Shouldn't we add bs to instances here?
instances.add(bs);

Reply:
agreed... done.